lms/djangoapps/course_goals/models.py

```diff
@@ -0,0 +1,32 @@
+"""
+Course Goals Models
+"""
+from django.contrib.auth.models import User
+from django.db import models
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from eventtracking import tracker
+from openedx.core.djangoapps.xmodule_django.models import CourseKeyField
+
+
+class CourseGoal(models.Model):  # pylint: disable=model-missing-unicode
+    """
+    Represents a course goal set by a user on the course home page.
+    """
+    user = models.ForeignKey(User)
+    course_key = CourseKeyField(max_length=255, db_index=True, blank=True)
+    goal_key = models.CharField(max_length=100, blank=True)
```

Comment:
What is this?

Reply:
The goal_key? I'll add a comment - it shows the goal, enumerated in CourseGoalOption - better to store a key that maps to an actual translated goal than store the english version of the goal string that could always change.

Reply:
Use `choices`: https://docs.djangoproject.com/en/1.11/ref/models/fields/#choices. Also, see https://pypi.python.org/pypi/django-choices.

Reply:
Adding now